corehq/apps/smsbillables/models.py

```diff
@@ -373,6 +375,13 @@ def _get_multipart_count(cls, backend_api_id, backend_instance, backend_message_
                 return int(segments)
             else:
                 raise RetryBillableTaskException("backend_message_id=%s" % backend_message_id)
+        elif backend_api_id == PinpointBackend.get_api_id():
+            pinpoint_message = get_pinpoint_message(backend_instance, backend_message_id)
+            segments = pinpoint_message['delivery']['numberOfMessageParts']
+            if segments is not None:
+                return int(segments)
+            else:
+                raise RetryBillableTaskException("backend_message_id=%s" % backend_message_id)
```

Comment:
Instead of checking another backend API ID and adding custom gateway logic here, this should be changed (including the above two branches for Twilio and Infobip) to check if the gateway has this feature, and if it does, delegate to the backend to get what it needs.